File: packages/api-models-ts/src/System.ts
/**
 * Request and response models for the instance introspection endpoints.
 *
 * @since 1.0.0
 */

import * as v from "valibot";

import * as Schemas from "./Schemas.ts";

/**
 * One parsed commit from the deployed SkyPortal git log.
 *
 * @since 1.0.0
 * @category Models
 */
export const GitLogEntry = Schemas.model(
    v.strictObject({
        time: Schemas.NullishString,
        sha: Schemas.NullishString,
        email: Schemas.NullishString,
        description: Schemas.NullishString,
        pr_nr: Schemas.NullishString,
        pr_url: Schemas.NullishString,
        commit_url: Schemas.NullishString,
        name: Schemas.NullishString,
    })
);

/**
 * @since 1.0.0
 * @category Models
 */
export type GitLogEntry = v.InferOutput<typeof GitLogEntry>;

/**
 * System and deployment information for the SkyPortal instance.
 *
 * @since 1.0.0
 * @category Models
 */
export const SysInfo = Schemas.model(
    v.strictObject({ gitlog: Schemas.list(GitLogEntry) })
);

/**
 * @since 1.0.0
 * @category Models
 */
export type SysInfo = v.InferOutput<typeof SysInfo>;

/**
 * Basic health information about the instance's database.
 *
 * @since 1.0.0
 * @category Models
 */
export const DbInfo = Schemas.model(
    v.strictObject({
        source_table_empty: Schemas.NullishBoolean,
        postgres_version: Schemas.NullishString,
    })
);

/**
 * @since 1.0.0
 * @category Models
 */
export type DbInfo = v.InferOutput<typeof DbInfo>;

File: packages/api-models-py/skyportal_py_models/system.py
"""Request and response models for the instance introspection endpoints."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class GitLogEntry(BaseModel):
    """One parsed commit from the deployed SkyPortal git log."""

    model_config = ConfigDict(extra="forbid")

    time: str | None = None
    sha: str | None = None
    email: str | None = None
    description: str | None = None
    pr_nr: str | None = None
    pr_url: str | None = None
    commit_url: str | None = None
    name: str | None = None


class SysInfo(BaseModel):
    """System and deployment information for the SkyPortal instance."""

    model_config = ConfigDict(extra="forbid")

    gitlog: list[GitLogEntry] = Field(default_factory=list)


class DBInfo(BaseModel):
    """Basic health information about the instance's database."""

    model_config = ConfigDict(extra="forbid")

    source_table_empty: bool | None = None
    postgres_version: str | None = None
